consensus/poet/core/sawtooth_poet/poet_consensus/utils.py

```diff
@@ -103,3 +105,19 @@ def build_certificate_list(block_header,
         LOGGER.error('Error getting block: %s', ke)
 
     return list(certificates)
+
+
+def state_view_for_block(block, state_view_factory):
+    """Returns a read-only state view object for a particular block
+
+    Args:
+        block (BlockWrapper): The block for which to return the state view.
+        state_view_factory (StateViewFactor): The state view factory
+            to be used to create the state view object
+
+    Returns:
+        StateView: The read-only state view for the block
+    """
+    state_root_hash = \
+        block.state_root_hash if block is not None else INIT_ROOT_KEY
```

Comment:
This `INIT_ROOT_KEY` stuff should maybe be pushed into the `create_view` function.  It feels like `INIT_ROOT_KEY` is leaking out here.  Pass in `None` as a root hash to `create_view`, get the view of `INIT_ROOT_KEY`.  In other words, users of the `StateViewFactory` shouldn't have to know about this detail.